lib/rubocop/cop/rails/action_order.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop enforces consistent ordering of the standard rails controller
+      # actions.
+      #
+      # The cop is configurable and can enforce any ordering of the standard
+      # actions. All order methods are ignored.
+      #
+      # [source,yaml]
+      # ----
+      #  Rails/ActionOrder:
+      #    ExpectedOrder:
+      #      - index
+      #      - show
+      #      - new
+      #      - edit
+      #      - create
+      #      - update
+      #      - destroy
+      # ----
+      #
+      # @example
+      #   # bad
+      #   def index; end
+      #   end destroy; end
+      #   def show; end
+      #
+      #   # good
+      #   def index; end
+      #   def show; end
+      #   end destroy; end
+      class ActionOrder < Base
+        MSG = 'Controller actions should be sorted in the following order: %<expected_order>s. ' \
+              'Action `%<current>s` should appear before `%<previous>s`.'
+
+        def_node_search :action_declarations, <<~PATTERN
+          (def {:index | :show | :new | :edit | :create | :update | :destroy} (args) ...)
```

Comment:
```suggestion
          (def {:index :show :new :edit :create :update :destroy} ...)
```

The bars are not necessary, and neither is the `args` node since you want to capture all of the `def` nodes regardless of arguments.